### PR TITLE
misc: don't set pass spec arguments from instance when they are the default value

### PR DESCRIPTION
**Pull request by superlopuh**
This is useful in a few places, it makes the `xdsl-gui` pass pipeline a little cleaner, as well as the marimo notebook showing the different stages of the pipeline in the slides.

### CODE REVIEW

File: xdsl/passes.py [hunk 1]
```diff
@@ -116,32 +116,39 @@ def required_fields(cls: type[ModulePassT]) -> set[str]:
             field.name for field in dataclasses.fields(cls) if not _is_optional(field)
         }
 
-    def pipeline_pass_spec(self) -> PipelinePassSpec:
+    def pipeline_pass_spec(self, *, include_default: bool = False) -> PipelinePassSpec:
         """
         This function takes a ModulePass and returns a PipelinePassSpec.
+
+        If `include_default` is `True`, then add arguments that are optional are not
+        included in the spec.
```
Comment (webmiche) [inline xdsl/passes.py:124]: This sentence is kinda hard for me to parse. I think I understand it, but how about this?

```suggestion
        If `include_default` is `True`, then optional arguments are not included in the spec.
```

File: tests/test_pass_to_spec.py
Comment (webmiche) [inline tests/test_pass_to_spec.py:1]: Why not split the tests to one testing with default and one without? This makes it much simpler and more straight-forward to understand what failed, if anything failed.

**Review by webmiche — APPROVED** [review]

**Review by dshaaban01 — APPROVED** [review]

File: tests/test_pass_to_spec.py
Comment (superlopuh) [reply tests/test_pass_to_spec.py:1]: Sure I'll do that

**Review by webmiche — APPROVED** [review]
LGTM!

**Pull request merged**